apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/JpaClaimRdaSink.java

```diff
@@ -0,0 +1,162 @@
+package gov.cms.bfd.pipeline.rda.grpc.sink;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import gov.cms.bfd.pipeline.rda.grpc.ProcessingException;
+import gov.cms.bfd.pipeline.rda.grpc.RdaChange;
+import gov.cms.bfd.pipeline.rda.grpc.RdaSink;
+import gov.cms.bfd.pipeline.sharedutils.PipelineApplicationState;
+import java.util.Collection;
+import javax.persistence.EntityExistsException;
+import javax.persistence.EntityManager;
+import lombok.Getter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * RdaSink implementation that writes TClaim objects to the database in batches.
+ *
+ * @param <TClaim> type of entity objects written to the database
+ */
+public class JpaClaimRdaSink<TClaim> implements RdaSink<RdaChange<TClaim>> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(JpaClaimRdaSink.class);
+
+  private final EntityManager entityManager;
+  private final Metrics metrics;
+
+  public JpaClaimRdaSink(String claimType, PipelineApplicationState appState) {
+    entityManager = appState.getEntityManagerFactory().createEntityManager();
+    metrics = new Metrics(appState.getMetrics(), claimType);
+  }
+
+  @Override
+  public void close() throws Exception {
+    if (entityManager != null && entityManager.isOpen()) {
+      entityManager.close();
+    }
+  }
+
+  public int writeBatch(Collection<RdaChange<TClaim>> claims) throws ProcessingException {
+    try {
+      metrics.calls.mark();
+      try {
+        persistBatch(claims);
```

Comment:
I'm thinking one day we might want to optimize this call differently for FISS and MCS records, I'm thinking thats achievable by using different batch sizes, in which case I think that's already built-in to your implementation, but wanted to double-check that each sink could be configured to use different batch sizes in the same env?

Reply:
Good question.  Currently both pipelines share the same settings for batch size, run interval, etc.  Can we defer that change to a later sprint?  I think load testing will reveal whether it's necessary or not.